src/core/raster/qgsrasterdataprovidertemporalproperties.h

```diff
@@ -0,0 +1,190 @@
+/***************************************************************************
+                         qgsrasterdataprovidertemporalproperties.h
+                         ---------------
+    begin                : February 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSRASTERDATAPROVIDERTEMPORALPROPERTIES_H
+#define QGSRASTERDATAPROVIDERTEMPORALPROPERTIES_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgsrange.h"
+#include "qgsdataprovidertemporalproperties.h"
+
+/**
+ * \class QgsRasterDataProviderTemporalProperties
+ * \ingroup core
+ * Implementation raster data provider temporal properties.
```

Comment:
```suggestion
 * Implementation of data provider temporal properties for QgsRasterDataProviders.
```

I'd extend these docs to describe the differences between the data provider temporal properties vs the layer temporal properties. Eg.

"Data provider temporal properties reflect the temporal capabilities of a QgsDataProvider. Unlike QgsMapLayerTemporalProperties, these settings are not user-configurable, and should only be set by the QgsDataProvider itself."

Actually, on that topic -- how would you feel about renaming this class QgsDataProviderTemporalCapabilties/QgsRasterDataProviderTemporalCapabilities?

Reply:
Thats okay with me.